### PR TITLE
Allow building with 9.2.4, add basic CI config

### PR DESCRIPTION
**Pull request by jhrcek**
The main goal is to allow this lib to be compiled with GHC 9.2.4

### CODE REVIEW

File: Readme.markdown [hunk 1]
```diff
@@ -138,24 +138,16 @@ cabal configure --enable-tests && cabal build -j
 ```
 
 To execute the tests, you need a running PostgreSQL server with an empty
-database called _test_. Tests are executed through cabal as follows:
+database called _test_. You can start one using docker, e.g.:
 
 ```bash
-cabal configure --enable-tests && cabal test
+docker run --rm -p 5432:5432 -e POSTGRES_HOST_AUTH_METHOD=trust -e POSTGRES_DB=test postgres:15-alpine
```
Comment (michivi) [inline Readme.markdown]: Should we also upgrade PostgreSQL in the CI file from 14 to 15 as well to stay consistent?

Comment (jhrcek) [reply Readme.markdown]: Good point, updated.

File: .github/workflows/haskell-ci.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        cabal:
+          - 3.6
+        ghc:
+          - 8.10.7
+          - 9.0.2
+          - 9.2.4
+
+    steps:
+      - uses: actions/checkout@v3
+        if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/master'
```
Comment (michivi) [inline .github/workflows/haskell-ci.yml]: Why are those conditions for this specific test? Were they actually meant as triggering events?

Comment (jhrcek) [reply .github/workflows/haskell-ci.yml]: TBH I don't have much idea what this means. I just copy pasted the config from elm-street and tweaked it. I think I'll just remove this line.

**Review by michivi — APPROVED** [review]
LGTM!

**Pull request merged**